### DISCUSSION TITLE
Alpine Vaultwarden is outdated and does not update to recent release

### DISCUSSION THREAD

**Discussion by johndoe0815** [tteck/Proxmox#2820] [Help Needed] [answered]
I have just installed alpine vaultwarden with your script and also ran it again to upgrade to latest version. 
As per 
https://pkgs.alpinelinux.org/packages?name=vaultwarden&branch=edge&repo=&arch=x86_64&maintainer=
it should install 1.30.5:
![grafik](https://github.com/tteck/Proxmox/assets/92877294/60a822c0-42a7-4805-a2ec-2fe384ec8bc9)

And as per
https://build.alpinelinux.org/buildlogs/build-edge-x86_64/community/vaultwarden/vaultwarden-1.30.5-r0.log 
it this would install the latest version with Web Vault Version 2024.1.2. However the web vault says that it is only on version 2023.10.0.
Diagnostics in Admin Panel is also saying that it is outdated:
![grafik](https://github.com/tteck/Proxmox/assets/92877294/b18cef83-ce1b-4de5-97ba-a18e15b76c91)

Why is it not updated to the latest release?

**Comment by tteck** [comment]
Are you using the edge branch or the 3.19 branch?

**Reply by tteck** [reply, marked as answer]
If you haven't adjusted the APK repositories to use `edge`, the script will utilize `latest-stable`.

**Reply by di5cord20** [reply]
How do I adjust the APK repositories to use edge?

**Reply by di5cord20** [reply]
Just asked ChatGPT and got my answer.

To adjust the Vaultwarden Alpine APK repository to use the `edge` branch, you need to update the `/etc/apk/repositories` file on your Alpine Linux installation. Here’s how to do it:

1. **Open the APK repositories file**:
   ```sh
   sudo nano /etc/apk/repositories
   ```

2. **Modify the repository URLs**:
   Replace the existing repository URLs with the `edge` URLs. For example, if you have:

   ```
   https://dl-cdn.alpinelinux.org/alpine/v3.18/main
   https://dl-cdn.alpinelinux.org/alpine/v3.18/community
   ```

   Change them to:

   ```
   https://dl-cdn.alpinelinux.org/alpine/edge/main
   https://dl-cdn.alpinelinux.org/alpine/edge/community
   https://dl-cdn.alpinelinux.org/alpine/edge/testing
   ```

   **Note**: Adding the `testing` repository is optional but may be useful if you need access to packages in the `testing` branch.

3. **Update APK and install the required packages**:
   After updating the repository list, run the following commands to update your system and install or upgrade Vaultwarden:

   ```sh
   sudo apk update
   sudo apk upgrade
   sudo apk add vaultwarden
   ```

4. **Verify the installation**:
   Confirm that Vaultwarden is installed or upgraded correctly by checking its version or status.

   ```sh
   vaultwarden --version
   ```

**Caution**: Using the `edge` repository is akin to using a rolling release, which may introduce instability. Ensure that this is suitable for your environment before proceeding.

This approach will point your Alpine Linux installation to the latest available packages in the `edge` branch.

**Reply by di5cord20** [reply]
My repositories are pointed to edge but the update script is not working.  Any ideas on how to solve?

**Reply by di5cord20** [reply]
Adding details from console and Diagnostics if helpful.

![efffbcf1-677d-402a-9240-6ee0f1e7d274](https://github.com/user-attachments/assets/7348867c-af3a-4577-b90a-df1fadc3389c)

![image](https://github.com/user-attachments/assets/00b95f6e-ea6e-4464-a0cc-829de16ba97c)


```fetch https://dl-cdn.alpinelinux.org/alpine/edge/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/edge/community/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/edge/testing/x86_64/APKINDEX.tar.gz
v20240807-524-g5e7f4223f0d [https://dl-cdn.alpinelinux.org/alpine/edge/main]
v20240807-541-g8f8d35db123 [https://dl-cdn.alpinelinux.org/alpine/edge/community]
v20240807-533-ge3e94180c65 [https://dl-cdn.alpinelinux.org/alpine/edge/testing]
OK: 32006 distinct packages available
OK: 105 MiB in 75 packages```

**Reply by navels** [reply]
The [update script](https://github.com/tteck/Proxmox/blob/main/ct/alpine-vaultwarden.sh) is just running
```
apk update && apk upgrade
```
so maybe open a shell and do some apk troubleshooting?

**Reply by di5cord20** [reply]
Any suggestions on what to troubleshoot?  I ran the apk update && apk upgrade manually as well... same results.

**Reply by tteck** [reply]
Try `rc-service vaultwarden restart`

**Reply by di5cord20** [reply]
thanks @tteck - I think that did the trick.  

![image](https://github.com/user-attachments/assets/024df1de-cf03-4972-9c2f-36234218bd8f)

**Reply by towerhand** [reply]
> Try `rc-service vaultwarden restart`

Is there a way to only restart the service if the vaultwarden was updated to a newer version?

**Reply by navels** [reply]
> > Try `rc-service vaultwarden restart`
> 
> Is there a way to only restart the service if the vaultwarden was updated to a newer version?

tteck already added the restart to the upgrade script.
